### PR TITLE
feat: add etherscan-source to cast

### PR DESCRIPTION
**Pull request by ayushm2003**
Ports `seth etherscan-source` to `cast`

`$ cast etherscan-source --chain <chain> <contract-address> --etherscan-api-key <api-key>`

Fixes required -

1. Escape sequence characters are not interpreted.
2. Probably better to use `NameOrAddress` type for contract address, but I was unable to figure it out.

Would appreciate any feedback as I am only starting out with rust.

### CODE REVIEW

**Comment by gakonst** [comment]
Hey - this is already implemented [here](https://github.com/gakonst/ethers-rs/blob/3b0b3132473dd72b3a8b5efbdb9b2917d9aa6c26/ethers-etherscan/src/contract.rs#L255)! Do you want to replace your logic with the one from this function?

File: cli/src/opts/cast.rs [hunk 1]
```diff
@@ -311,6 +311,16 @@ pub enum Subcommands {
...
+    #[structopt(name = "etherscan-source")]
+    #[structopt(about = "Prints the source code of a contract from Etherscan")]
+    EtherscanSource {
+        #[structopt(long, env = "CHAIN", default_value = "mainnet")]
+        chain: Chain,
+        #[structopt(help = "the contract address")]
+        address: String,
+        #[structopt(long, env = "ETHERSCAN_API_KEY")]
+        etherscan_api_key: String,
+    },
```
Comment (gakonst) [inline cli/src/opts/cast.rs:323]: this looks good

**Comment by ayushm2003** [comment]
> Hey - this is already implemented [here](https://github.com/gakonst/ethers-rs/blob/3b0b3132473dd72b3a8b5efbdb9b2917d9aa6c26/ethers-etherscan/src/contract.rs#L255)! Do you want to replace your logic with the one from this function?

cool, I'll do it.

File: cast/src/lib.rs [hunk 2]
```diff
@@ -852,6 +853,38 @@ impl SimpleCast {
...
+    ///     );
+    /// #    Ok(())
+    /// # }
+    /// ```
+    pub async fn etherscan_source(
+        chain: Chain,
+        contract_address: String,
+        etherscan_api_key: String,
+    ) -> Result<String> {
+        let client = Client::new(chain, etherscan_api_key).unwrap();
```
Comment (gakonst) [inline cast/src/lib.rs:878]:
```suggestion
        let client = Client::new(chain, etherscan_api_key)?;
```

File: cast/src/lib.rs [hunk 3]
```diff
@@ -852,6 +853,38 @@ impl SimpleCast {
...
+        let meta = client.contract_source_code(contract_address.parse().unwrap()).await.unwrap();
```
Comment (gakonst) [inline cast/src/lib.rs:879]:
```suggestion
        let meta = client.contract_source_code(contract_address.parse()?).await?;
```

**Review by gakonst — APPROVED** [review]
let's avoid unwraps? lgtm otherwise

File: cast/src/lib.rs (hunk 2 above)
Comment (ayushm2003) [reply cast/src/lib.rs:878]: Yeah cool. I was under the impression that they both are the same thing but just realized that `unwrap()` panics, so makes sense.

**Review by gakonst — APPROVED** [review]

**Comment by gakonst** [comment]
Just need to fix the lint errors and good to go!

**Comment by ayushm2003** [comment]
This is weird. Tests run fine on my machine, including clippy. Would you like me to make any changes?

**Comment by gakonst** [comment]
All good - had to rebase and add the import

**Pull request merged**